servicetalk-http-utils/src/main/java/io/servicetalk/http/utils/RedirectingHttpRequesterFilter.java

```diff
@@ -117,13 +150,30 @@ public RedirectingHttpRequesterFilter(final boolean onlyRelativeClient,
      * @param onlyRelativeClient Limits the redirects to relative paths for {@link HttpClient} filters.
      * @param onlyRelativeConnection Limits the redirects to relative paths for {@link HttpConnection} filters.
      * @param maxRedirects The maximum number of follow up redirects.
+     * @deprecated Use {@link Builder}.
      */
+    @Deprecated
     public RedirectingHttpRequesterFilter(final boolean onlyRelativeClient,
                                           final boolean onlyRelativeConnection,
                                           final int maxRedirects) {
-        this.onlyRelativeClient = onlyRelativeClient;
-        this.onlyRelativeConnection = onlyRelativeConnection;
-        this.maxRedirects = maxRedirects;
+        this(maxRedirects, onlyRelativeClient);
+    }
+
+    private RedirectingHttpRequesterFilter(final int maxRedirects,
```

Comment:
Perhaps using the existing public constructor would reduce the total amount of constructors as in the end that public constructor can be transformed to a private one. Also, the other new private constructor could flip the order of arguments to reduce the ambiguity level that having first boolean paramter mean different things depending on the chosen constructor.